src/bindings/python/src/pyopenvino/core/common.cpp

```diff
@@ -37,35 +37,37 @@ py::dtype get_dtype(const ov::element::Type& ov_type) {
     return ov_type_to_dtype().at(ov_type);
 }
 
-const std::map<int, ov::element::Type>& dtype_num_to_ov_type() {
-    static const std::map<int, ov::element::Type> dtype_to_ov_type_mapping = {
-        {23, ov::element::f16},  // float16
-        {11, ov::element::f32},  // float32
-        {12, ov::element::f64},  // float64
-        {1, ov::element::i8},    // int8
-        {3, ov::element::i16},   // int16
-#ifdef _WIN32
-        {7, ov::element::i32},  // int32
-        {9, ov::element::i64},  // int64
-#else
-        {5, ov::element::i32},  // int32
-        {7, ov::element::i64},  // int64
-#endif
-        {2, ov::element::u8},   // uint8
-        {4, ov::element::u16},  // uint16
-#ifdef _WIN32
-        {8, ov::element::u32},   // uint32
-        {10, ov::element::u64},  // uint64
-#else
-        {6, ov::element::u32},  // uint32
-        {8, ov::element::u64},  // uint64
-#endif
-        {0, ov::element::boolean},  // bool
-        {18, ov::element::string},  // bytes_
-        {19, ov::element::string},  // str_
-        {18, ov::element::string},  // bytes
-        {19, ov::element::string},  // str
+std::map<int, ov::element::Type> init_num_to_ov_type() {
+    static const std::map<std::string, ov::element::Type> str_to_type_mapping = {
```

Comment:
should it be static in this case?

Reply:
Why shouldn't it be?

Reply:
`init_num_to_ov_type` is called once. And `str_to_type_mapping` is need only for internal usage so we don't need to keep it for the lifetime of the project.

Reply:
After discussion with @praasz there is no huge difference in using "static" or not using it in this case.
What we can put in "TODO" is refactoring the std::map into std::array of pairs like <char*, Type>.

Reply:
Static local variable, once initialized, occupies memory till end of a program. That makes sense when function is public and used repeatedly. In current scenario it's complicated ;) .. it's public but used only once.

Reply:
It could be refactored together with https://github.com/openvinotoolkit/openvino/pull/24626#discussion_r1609805268 -- thanks for spotting this! @akuporos @t-jankowski @praasz